### ISSUE TITLE
Separate serializers for different endpoints

### ISSUE THREAD

**Issue by TehMillhouse** [jfinkels/flask-restless#440]
Currently, serializers just flat out don't work for anything but vanilla `GET` requests (see #413, #422). A pattern seen relatively often in the wild for REST interfaces is to provide only a listing of objects for the `GET_MANY` endpoint and only serializing all fields of an object for the `GET_SINGLE` endpoint.  
i.e., something like `GET /api/orders` would return:
```json
{
  "num_results": 2,
  "objects": [
    {
      "id": 1
    },
    {
      "id": 2
    }
  ],
  "page": 1,
  "total_pages": 1
}
```

While `GET /api/orders/1` would return:
```json
{
  "id": 1,
  "name": "Customer McBuy",
  "item_ids": [
    13,
    52,
    7
  ]
}
```

It would be very nice to be able to do this with Flask-ReSTless without taking a huge performance hit, however the current custom serializer support is very limited. My proposition is:  
introduce a new keyword parameter to `create_api` called `serializers` which works just like `pre`/`postprocessors`: a dict which specifies for every version of every endpoint which serializer is to be used. (optionally deprecate the current `serializer` argument (as it's broken) or use it as fallback for endpoints not specified in `serializers`)

This would take care of #413 and #422 and incidentally scratch my own itch :)

**Comment by jfinkels** [comment]
The code has been completely rewritten to support the JSON API protocol, so I'm not sure if this issue still applies. Related to #475 as well.

**Comment by jfinkels** [comment]
This should be fixed by pull request #493.
